alibi/explainers/shap_wrappers.py

```diff
@@ -0,0 +1,1431 @@
+        """
+
+        if not self._fitted:
+            raise TypeError(
+                "Called explain on an unfitted object! Please fit the "
+                "explainer using the .fit method first!"
+            )
+
+        self.tree_limit = tree_limit
+        self.approximate = approximate
+        self.interactions = interactions
+        self.summarise_result = summarise_result
+
+        if interactions:
+            self._check_interactions(approximate, self.background_data, y)
+            shap_output = self._explainer.shap_interaction_values(X, tree_limit=tree_limit)
+        else:
+            self._check_explainer_setup(self.background_data, self.model_output, y)
+            shap_output = self._explainer.shap_values(
+                X,
+                y=y,
+                tree_limit=tree_limit,
+                approximate=self.approximate,
+                check_additivity=check_additivity,
+            )
+        expected_value = self.expected_value
+        if isinstance(shap_output, np.ndarray):
+            shap_output = [shap_output]
+        if isinstance(expected_value, float):
+            expected_value = [self.expected_value]
+
+        explanation = self.build_explanation(
+            X,
+            shap_output,
+            expected_value,
+            summarise_result=summarise_result,
+            cat_vars_start_idx=cat_vars_start_idx,
+            cat_vars_enc_dim=cat_vars_enc_dim,
+        )
+
+        self._update_metadata(
+            {'interactions': self.interactions,
+             'explain_loss': True if y is not None else False,
+             'approximate': self.approximate,
+             },
+            params=True,
+        )
+
+        return explanation
+
+    def _check_interactions(self, approximate: bool, background_data, y: Optional[np.ndarray]) -> None:
+        """
+        Checks if the inputs to the explain method match the explainer setup if shap interaction values
+        are to be explained.
+
+        Parameters
+        ----------
+        approximate
+            See `explain` documentation.
+        background_data
+            See `fit` documentation.
+        y
+            See explain documentation.
+
+        Raises
+        ------
+        NotImplementedError
+            If a background dataset is passed to the `fit` method or argument `y` is specified to the `explain`
+            method. These algorithms are not yet supported in the shap library.
+
+        Warns
+        -----
+            If approximate values are requested. These values are not defined for interactions.
+        """
+
+        self.approximate = approximate
+        if approximate:
+            logger.warning("Approximate shap values are not defined for shap interaction values, "
+                           "ignoring argument!")
+            self.approximate = False
+        if background_data is not None:
+            # TODO: @Janis: Here we can just disable the interactions, give a warning and just
```

Comment:
I think I generally prefer raising an error rather than giving a warning (which can be disabled and thus unseen by a user) and doing a different thing than what was requested.

Reply:
ok, let's do it whis way then!